### DISCUSSION TITLE
Rolladensteuerung

### DISCUSSION THREAD

**Discussion by kuchn7331** [Zefau/ioBroker.jarvis#2377] [Q&A] [answered]
Hey Leute,
ich bin seit Stunden am googeln und probieren, aber ich bekomme es nicht hin.
Ich hab unsere Rolläden mit einem Zigbeeaktor ausgestattet, welcher über zigbee2mqtt mit dem iobroker kommuniziert.
Jetzt wollte ich die Rolläden auf die Jarvis Oberfläche bringen. Aber irgendwie ist da der Wurm drin.

Ich lege die Geräte manuell an, als Gewerk nehme ich Rolläden, Jalousie. Trage den Datenpunkt aus dem Objekte Ordner ein. Das Problem ist, dass die Daten von zigbee2mqtt im json Format übertragen werden.
so z.b.
{
  "calibration": "OFF",
  "calibration_time": 15.8,
  "linkquality": 145,
  "motor_reversal": "OFF",
  "moving": "STOP",
  "position": 50,
  "state": "OPEN"
}
das steuern des Aktors erfolgt direkt aus dem ioB sehr gut, indem ich den z.b. Datenpunkt mqtt.0.zigbee2mqtt.Rollo.set  WERT: {"state": "OPEN"} ändere.
Aber aus Jarvis heraus, schreibt der Adapter den Wert aus als Zahlenwert bzw. Wortwert und nicht als Json String.
Ich hoffe ihr versteht was ich meine.

Wie bekomme ich jetzt Jarvis dazu, den Wert im korrekten Format zu schreiben?

Vielen Dank schon mal!

**Comment by mcuiobroker** [comment]
Bitte die ioBroker-DPs zu dem Rollladen.
Und eine Beschreibung was du wo einträgst damit eine Funktion ausgeführt wird.

Datenpunkt-Eigenschaften (Versuch)
```
{"on":{\"state\":\"OPEN\"},"off":{\"state\":\"CLOSE\"}}
```

**Reply by kuchn7331** [reply]
![Screenshot_20230529_232730_Chrome.jpg](https://github.com/Zefau/ioBroker.jarvis/assets/134972730/a2fcf003-baec-44e9-a529-be272dbccde7)

Bei RolloHWR kommen die Daten im json Format.
Als Unterpunkt habe ich RolloHWR.set angelegt.
Dort kann ich die Werte { "state":"OPEN"}
Oder CLOSE oder STOP eingeben, welches vom aktor dann auch ausgeführt wird.

**Comment by mcuiobroker** [comment]
Wie möchtest du es denn in jarvis steuern? DropDownAction?
Welchen Type hat der DP?
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/c0d964ba-ffbc-4e14-8f08-c3a7c27972ec)


![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/08da79fc-0e8a-426f-9f82-5088868db507)
Anzeigevariante
```
{"{\"state\":\"OPEN\"}":"Open","{\"state\":\"STOP\"}":"Stop","{\"state\":\"CLOSE\"}":"Close"}
```

**Reply by kuchn7331** [reply]
Das wäre mir grundsätzlich egal. Cool wäre natürlich das vom Bild unten, aber ich nehme das was möglich ist. 

Bei dem Objektdaten steht Typ: string/Zeichenkette 

Ich teste mal die Vorschläge. 

![Screenshot_20230530_064814_Chrome.jpg](https://github.com/Zefau/ioBroker.jarvis/assets/134972730/fad09324-a371-4386-be9a-d97b0b3163b1)

**Comment by kuchn7331** [comment]
hello again,

also das hier hat schon mal sehr gut geklappt. über das dropdownmenu gehts es erstmal.

> {"{\"state\":\"OPEN\"}":"Open","{\"state\":\"STOP\"}":"Stop","{\"state\":\"CLOSE\"}":"Close"}

allerdings bekomme ich das mit dem slider nicht hin. Der Wert müsste wieder lauten : {"position": VALUE}
aber wo genau müsste ich das einfügen . belasse ich den slider wie er ist, sendet er mir nur einen Rohwert.
Wie bekomme ich das noch hin?

**Reply by mcuiobroker** [reply]
Um das nutzen zu können müsstest du sehr wahrscheinlich ein eigenes Script schreiben.
Jarvis kann die Daten ??? nicht aus einem "Verzeichnis" lesen??
Es geht um den "DP" ROLLOHWR.
Steht da immer das gleiche drin? Welche Daten stehen da drin?

**Reply by kuchn7331** [reply]
![iob](https://github.com/Zefau/ioBroker.jarvis/assets/134972730/98cda709-3e8c-4e70-aad1-4fdb792f4403)

so sieht der Datenpunkt des RolloHWR aus. Die werte kann man aber nicht ändern.
Steuern lässt sich das Gerät nur indem man ein payload über mqtt published, wie oben schon erwähnt.
Gesendet wird das an den Datenpunkt mit Unterpunkt .set .
Oder man ändert den Wert halt direkt über die Objekte im Iobroker.
hier stehts noch mal im einzelnen: [https://www.zigbee2mqtt.io/devices/QS-Zigbee-C03.html#lonsonho-qs-zigbee-c03](url)

für das anfahren einer position müsste ich jarvis praktisch dazu bringen den datenpunkt .set mot dem: {"position": 1-100} zu überschreiben.

**Comment by mcuiobroker** [comment]
Wie hast du die DPs für den Rollladen in jarvis angelegt? Zeig mal bitte die 3. level, activity, stop

**Reply by kuchn7331** [reply]
![iob](https://github.com/Zefau/ioBroker.jarvis/assets/134972730/1908730f-35c7-4b37-a8c3-fc7b648f07e3)

da hab ich bisher nur level drin.

**Comment by mcuiobroker** [comment]
@Zefau Siehst du eine Möglichkeit bei diesem Rollladen (ZigbeeMQTT) die Einstellungen für die BlindLevelAction und dessen DPs so einzustellen, dass man den LevelBody nutzen kann?
Gibt es für den BlindLevel die Option "position"? Und könnte man diesen als Objektstring (siehe oben) angeben und parsen?

Ich habe es versucht komme aber nicht weiter. Man müsste es mit einem zusätzlichem Script umsetzen und eigene DPs erstellen.

**Comment by mcuiobroker** [comment]
Was steht in dem JSON, wenn der Rollladen sich bewegt.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/d1f3ea16-8bb3-4a03-a09a-0b62adc0e873)
Wie oft werden die Daten aktualisiert? 1*/sec?

**Comment by mcuiobroker** [comment, marked as answer]
Hab es in die Doku gesetzt:
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis-v3/system-effektprog./blindleveljson

**Comment by kuchn7331** [comment]
Sorry für die späte Antwort.
Hab es getestet und funzt wunderbar.
Vielen Dank! :)

**Comment by kuchn7331** [comment]
hey,
ich nochmal. ich könnte schwören gestern mit den test dp,s ging alles.
heute habe ich das ganze auf mein Rolle umschreiben wollen, ging nicht.
also alles wieder rückgängig, geht auch nicht mehr. dafür bekomme ich beim betätigen in jarvis eine ganze reihe warnmeldungen im log.

<html>
<body>


javascript.0 | 2023-06-05 22:18:26.266 | warn | at processImmediate (node:internal/timers:466:21)
-- | -- | -- | --
javascript.0 | 2023-06-05 22:18:26.265 | warn | at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.js:5916:56)
javascript.0 | 2023-06-05 22:18:26.265 | warn | at Object.stateChange (/opt/iobroker/node_modules/iobroker.javascript/main.js:593:29)
javascript.0 | 2023-06-05 22:18:26.265 | warn | at Object.callback (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1244:38)
javascript.0 | 2023-06-05 22:18:26.265 | warn | at Object.<anonymous> (script.js.common.blindlevelJSON:58:9)
javascript.0 | 2023-06-05 22:18:26.265 | warn | at setState (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1719:20)
javascript.0 | 2023-06-05 22:18:26.263 | warn | State "" not found


</body>
</html

ich werde hier noch verrückt. ich hab schon alles nochmal überprüft, neu gestartet, script neu erstellt nach vorlage. ich kriegs nicht mehr hin.

**Reply by mcuiobroker** [reply]
Wie sieht denn das script bei dir jetzt aus, stell es mal rein.
Was hast du wo umgestellt?
Im script wird nichts umgestellt, nur in den Objekten!
Nach der Fehlerbeschreibung ist der mqttsetDP nicht gesetzt.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/a4f3bc6b-30fb-4b2c-b97f-7274d1e982e4)
Zeig mal Deine Objekte

**Comment by kuchn7331** [comment]
hey,
ich hatte nen tippfehler, genau wie du sagtest im set DP. sorry dafür.
Muss ich dann für jedes Rolle ein extra script machen, oder kann man alles in eines rein packen? was müsste ich da anpassen?
danke für die tolle unterstützung! (:

**Comment by mcuiobroker** [comment]
Du musst dann mehrere Scripte erstellen und dort in den Scripten die eine Zeile ändern. Sonst wie gehabt.
`let blindLevelJsonDP    = '0_userdata.0.jarvis.blindLevelJson'; `// Kann man für unterschiedliche Tastern erweitern -> `blindLevelJSON1`

`let blindLevelJsonDP    = '0_userdata.0.jarvis.blindLevelJson1';` // usw. bei jedem Script halt anders, damit man es in Jarvis unterscheiden kann.
